[libsrtp] Build error - cisco/libsrtp/pull/573.diff does not have the expected hash

Package: libsrtp:x64-windows -> 2.4.2#3

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.37.32822.0
-    vcpkg-tool version: 2023-08-09-9990a4c9026811a312cb2af78bf77f3d9d288416
    vcpkg-scripts version: d5b03c125 2023-08-21 (8 hours ago)

**To Reproduce**

`vcpkg install libsrtp`
**Failure logs**

```
-- Downloading https://patch-diff.githubusercontent.com/raw/cisco/libsrtp/pull/573.diff -> libsrtp-pr-573.patch...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Trying to load bundleconfig from C:\Users\gerard\git\immersaview\sender\vcpkg-overlay\vcpkg\vcpkg-bundle.json
[DEBUG] Failed to open: C:\Users\gerard\git\immersaview\sender\vcpkg-overlay\vcpkg\vcpkg-bundle.json
[DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[DEBUG] Metrics enabled.
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Feature flag 'dependencygraph' unset
Downloading https://patch-diff.githubusercontent.com/raw/cisco/libsrtp/pull/573.diff
[DEBUG] Trying to hash C:\Users\gerard\git\immersaview\sender\vcpkg-overlay\vcpkg\downloads\libsrtp-pr-573.patch.1740.part
[DEBUG] C:\Users\gerard\git\immersaview\sender\vcpkg-overlay\vcpkg\downloads\libsrtp-pr-573.patch.1740.part has hash 78a1e41dd4f05e9e9fae1b47cc73e0f07bb5bfab4118c7f22b764487a3bf5f7e0ca9405aff026c075faa94033b68e85ad684261c6142751ef426524d5968ed5e
error: Failed to download from mirror set
error: File does not have the expected hash:
url: https://patch-diff.githubusercontent.com/raw/cisco/libsrtp/pull/573.diff
File: C:\Users\gerard\git\immersaview\sender\vcpkg-overlay\vcpkg\downloads\libsrtp-pr-573.patch.1740.part
Expected hash: 58c07977ccbd3dce114f59583a5e10813bbee907af5ef23ec1961b4cfd0791c14326845a376470430978e878c78fb1f598851898a9e9031225628739fb248176
Actual hash: 78a1e41dd4f05e9e9fae1b47cc73e0f07bb5bfab4118c7f22b764487a3bf5f7e0ca9405aff026c075faa94033b68e85ad684261c6142751ef426524d5968ed5e
[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(1053): 
[DEBUG] Time in subprocesses: 0us
[DEBUG] Time in parsing JSON: 18us
[DEBUG] Time in JSON reader: 0us
[DEBUG] Time in filesystem: 1662us
[DEBUG] Time in loading ports: 0us
[DEBUG] Exiting after 485 ms (465514us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
      
      Failed to download file with error: 1
      If you are using a proxy, please check your proxy setting. Possible causes are:
      
      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.
      
      2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
         set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
         The value set by your proxy might be wrong, or have same `https://` prefix issue.
      
      3. Your proxy's remote server is out of service.
      
      If you've tried directly download the link, and believe this is not a temporary
      download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
      to report this upstream download server failure.
      

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
  ports/libsrtp/portfile.cmake:1 (vcpkg_download_distfile)
  scripts/ports.cmake:147 (include)



```

--- a/ports/libsrtp/portfile.cmake
+++ b/ports/libsrtp/portfile.cmake
@@ -1,16 +1,9 @@
-vcpkg_download_distfile(CMAKE_PR_PATCH
-    URLS https://patch-diff.githubusercontent.com/raw/cisco/libsrtp/pull/573.diff
-    FILENAME libsrtp-pr-573.patch
-    SHA512 58c07977ccbd3dce114f59583a5e10813bbee907af5ef23ec1961b4cfd0791c14326845a376470430978e878c78fb1f598851898a9e9031225628739fb248176
-)
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO cisco/libsrtp
-    REF v2.4.2
-    SHA512 6E4805E6D34B2050A6F68F629B0B42356B1D27F2CBAA6CC6166E56957609C3D9AA6B723DCC674E5C74180D122D27BADD2F9496639CCB1E0C210B9E1F7949D0E2
-    PATCHES 
-        ${CMAKE_PR_PATCH}
+    REF "v${VERSION}"
+    SHA512 bd679ab65ccf22ca30fe867b9649a0b84cfa6fad6e22eb10f081141632f6dd56479a04d525b865f11fd46007303ca211065d9c170e4820d6ea7055403702340a
+    PATCHES
         fix-runtime-destination.patch
 )
 
@@ -18,14 +11,13 @@ vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         "-DCMAKE_PROJECT_INCLUDE=${CMAKE_CURRENT_LIST_DIR}/cmake-project-include.cmake"
+        -DTEST_APPS=OFF
 )
 
 vcpkg_cmake_install()
 
-vcpkg_cmake_config_fixup(
-    CONFIG_PATH lib/cmake/libSRTP
-)
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/libSRTP)
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/libsrtp" RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

--- a/ports/libsrtp/vcpkg.json
+++ b/ports/libsrtp/vcpkg.json
@@ -1,10 +1,9 @@
 {
   "name": "libsrtp",
-  "version": "2.4.2",
-  "port-version": 3,
+  "version": "2.5.0",
   "description": "This package provides an implementation of the Secure Real-time Transport Protocol (SRTP), the Universal Security Transform (UST), and a supporting cryptographic kernel.",
   "homepage": "https://github.com/cisco/libsrtp/",
-  "license": null,
+  "license": "BSD-3-Clause",
   "dependencies": [
     {
       "name": "vcpkg-cmake",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4649,8 +4649,8 @@
       "port-version": 3
     },
     "libsrtp": {
-      "baseline": "2.4.2",
-      "port-version": 3
+      "baseline": "2.5.0",
+      "port-version": 0
     },
     "libssh": {
       "baseline": "0.10.5",

--- a/versions/l-/libsrtp.json
+++ b/versions/l-/libsrtp.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "27c325c9851b5c07e172e74e312e8070af7b9d0b",
+      "version": "2.5.0",
+      "port-version": 0
+    },
+    {
       "git-tree": "d66d808ef040f2802ffa00ec6c4728d5603e76f9",
       "version": "2.4.2",
       "port-version": 3

